src/world.cpp

```diff
@@ -103,36 +124,54 @@ World::update_all_chunks_mesh() {
         }
     }
 
-    for (size_t i = 0; i < num_chunks; i++)
-        update_single_mesh(i);
+    GlobalContext& context = GlobalContext::getInstance();
+    for (size_t chunk_pos = 0; chunk_pos < num_chunks; chunk_pos++) {
+        context.push_task(
+            [this](ChunkIndex p) { this->update_single_mesh(p); }, chunk_pos
+        );
+    }
 
-    LOG_DEBUG(logging::terrain_logger, "End load chunks mesh");
+    // should only wait for the previously queued tasks, but I'm lazy and don't
+    // want to implement that until it will actually be used.
+    context.wait_for_tasks();
+    send_updated_chunks_mesh();
```

Comment:
this is unnecessary to lock on if you wait you can just do this directly in your task

Reply:
No. That is not the only place where it will be called.